packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -224,14 +225,48 @@ export class FieldSchema<
 	protected _typeCheck?: MakeNominal;
 
 	/**
-	 * @param kind - The [kind](https://en.wikipedia.org/wiki/Kind_(type_theory)) of this field.
-	 * Determine the multiplicity, viewing and editing APIs as well as the merge resolution policy.
-	 * @param allowedTypes - What types of tree nodes are allowed in this field.
+	 * Lazily evaluated allowed types.
 	 */
+	private readonly lazyTypes: Lazy<AllowedTypes>;
+
+	/**
+	 * What types of tree nodes are allowed in this field.
+	 *
+	 * @remarks
+	 * This is the same set of types in {@link FlexFieldSchema.allowedTypes}, just as a set with
+	 * laziness removed.
+	 */
+	public get allowedTypeSet(): AllowedTypes {
+		return this.lazyTypes.value;
+	}
+
 	public constructor(
+		/**
+		 * The {@link https://en.wikipedia.org/wiki/Kind_(type_theory) | kind } of this field.
+		 * Determines the multiplicity, viewing and editing APIs as well as the merge resolution policy.
+		 */
 		public readonly kind: Kind,
+		/**
+		 * What types of tree nodes are allowed in this field.
+		 */
 		public readonly allowedTypes: Types,
-	) {}
+	) {
+		this.lazyTypes = new Lazy(() => FieldSchema.normalizeAllowedTypes(this.allowedTypes));
+	}
+
+	/**
+	 * Normalizes a {@link ImplicitAllowedTypes} to a {@link AllowedTypes}.
+	 */
+	public static normalizeAllowedTypes(types: ImplicitAllowedTypes): AllowedTypes {
```

Comment:
This doesn't seem like something we expect apps to use directly, and also logically isn't really specific to FieldSchema. We can address both of these by moving it to a free function which we don't package export as public.